cmake/ThrustUtilities.cmake

```diff
@@ -0,0 +1,12 @@
+# Given a cu_file (e.g. foo/bar.cu) relative to CMAKE_CURRENT_SOURCE_DIR
+# and a thrust_target, create a cpp file that includes the .cu file, and set
+# ${cpp_file_var} in the parent scope to the full path of the new file. The new
+# file will be generated in:
+# ${CMAKE_CURRENT_BINARY_DIR}/<thrust_target_prefix>/${cu_file}.cpp
+function(thrust_wrap_cu_in_cpp cpp_file_var cu_file thrust_target)
+  thrust_get_target_property(prefix ${thrust_target} PREFIX)
+  set(wrapped_source_file "${CMAKE_CURRENT_SOURCE_DIR}/${cu_file}")
+  set(cpp_file "${CMAKE_CURRENT_BINARY_DIR}/${prefix}/${cu_file}.cpp")
+  configure_file("${Thrust_SOURCE_DIR}/cmake/wrap_source_file.cpp.in" "${cpp_file}")
+  set(${cpp_file_var} "${cpp_file}" PARENT_SCOPE)
```

Comment:
The tl;dr behind this is that we can't set the "treat as cpp" flag as a source file property anymore, since each source file is now compiled multiple times for multiple targets, some of which may still need to treat the file as CUDA.

Instead, a `.cpp` file is generated that just does `#include <thrust/original/file.cu>`, and that cpp file is compiled instead of the .cu.